api/scpca_portal/management/commands/load_data.py

```diff
@@ -59,110 +50,197 @@ class Command(BaseCommand):
+        """
+        project_path = common.INPUT_DATA_PATH / project_metadata["scpca_project_id"]
+        if project_path not in common.INPUT_DATA_PATH.iterdir():
+            logger.warning(
+                f"Metadata found for {project_metadata['scpca_project_id']},"
+                "but no s3 folder of that name exists."
+            )
+            return False
+
+        if project_metadata["pi_name"] not in whitelist:
+            logger.warning("Project submitter is not in the white list.")
+            return False
+
+        return True
+
+    def purge_project(
+        self,
+        project: Project,
+        *,
+        reload_existing: bool = False,
+        update_s3: bool = False,
+    ) -> bool:
+        """
+        Purges existing projects from the db so that they can be re-added when they are processed.
+        S3 is updated accordingly. Returns boolean as success status.
+        """
+        # Projects can only be intentionally purged.
+        # If the reload_existing flag is not set, then the project should not be procssed.
+        if not reload_existing:
+            logger.info(f"'{project}' already exists. Use --reload-existing to re-import.")
+            return False
+
+        # Purge existing projects so they can be re-added.
+        logger.info(f"Purging '{project}")
+        project.purge(delete_from_s3=update_s3)
+        return True
+
+    def create_computed_file(self, future, *, update_s3: bool, clean_up_output_data: bool) -> None:
+        """
+        Saves computed file returned from future to the db.
+        Uploads file to s3 and cleans up output data depending on passed options.
+        """
+        if computed_file := future.result():
+
+            # Only upload and clean up projects and the last sample if multiplexed
+            if computed_file.project or computed_file.sample.is_last_multiplexed_sample:
+                if update_s3:
+                    s3.upload_output_file(computed_file.s3_key, computed_file.s3_bucket)
+                if clean_up_output_data:
+                    computed_file.clean_up_local_computed_file()
+            computed_file.save()
+
+        # Close DB connection for each thread.
+        connection.close()
 
     @staticmethod
-    def clean_up_input_data(project):
+    def clean_up_input_data(project) -> None:
         shutil.rmtree(common.INPUT_DATA_PATH / project.scpca_id, ignore_errors=True)
 
     @staticmethod
-    def clean_up_output_data():
+    def clean_up_output_data() -> None:
         for path in Path(common.OUTPUT_DATA_PATH).glob("*"):
             path.unlink(missing_ok=True)
 
     def load_data(
         self,
-        allowed_submitters: set[str] = None,
-        input_bucket_name: str = common.INPUT_BUCKET_NAME,
+        input_bucket_name,
+        clean_up_input_data,
+        clean_up_output_data,
+        max_workers,
+        reload_existing,
+        scpca_project_id,
+        update_s3,
+        whitelist,
```

Comment:
Let's add the type hints back in. Also, I think that we should change `ALLOWES_SUBMITTERS` to `WHITELIST` to match.